test/files/run/t10783.scala

```diff
@@ -0,0 +1,24 @@
+package com.example {
+  object X {
+    def bar: Int = (new Value(42)).foo
+  }
+}
+
+package com.example {
+  class Value(val value: Int) extends AnyVal {
+    def foo: Int = value + 1
+  }
+}
```

Comment:
Test variations with an explicitly written companion, defined before and after this class, would be welcome.